internal/stolonctl/pkg/cluster/upgrade.go

```diff
@@ -93,11 +93,8 @@ func Upgrade(ctx context.Context, config Config) error {
 		return trace.Wrap(err)
 	}
 
-	objMeta := metav1.ObjectMeta{
-		Name:      defaults.CRDName,
-		Namespace: config.Namespace,
-	}
-	res, err := crdclient.CreateOrRead(objMeta)
+	resourceName := fmt.Sprintf("%s-%s", defaults.CRDName, config.Upgrade.Changeset)
```

Comment:
not absolutely necessary, but this resourceName format looks like it appears in a couple places, might be worth considering having a helper function that takes CRDName and Changeset and returns the resourceName. Just makes it easier to find all the paths that use that name, if you need to change it for any reason in the future.